kernel/timeout.c

```diff
@@ -90,20 +90,21 @@ void z_add_timeout(struct _timeout *to, _timeout_func_t fn,
 	__ASSERT_NO_MSG(arch_mem_coherent(to));
 #endif
 
-	k_ticks_t ticks = timeout.ticks + 1;
-
-	if (IS_ENABLED(CONFIG_TIMEOUT_64BIT) && Z_TICK_ABS(ticks) >= 0) {
-		ticks = Z_TICK_ABS(timeout.ticks) - (curr_tick + elapsed());
-	}
```

Comment:
Good catch.  Indeed, this computation should always have happened inside the lock, anyway.